echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/monitor/TriggerMonitor.java

```diff
@@ -77,6 +78,10 @@ private void triggerMatchingPipelines(T event) {
       List<Pipeline> allPipelines = pipelineCache.getPipelinesSync();
       List<Pipeline> matchingPipelines = eventHandler.getMatchingPipelines(event, allPipelines);
       matchingPipelines.stream()
+        .map(pipelineCache::refresh) // make sure we don't start an execution of a stale pipeline config
+        .map(p -> eventHandler.withMatchingTrigger(event, p)) // there is a possibility the refreshed pipeline no longer matches the trigger
```

Comment:
It looks like this is going to cause us to re-do all the trigger matching logic twice on every pipeline start.  In a lot of cases, there is a reasonable amount of logic in `withMatchingTrigger`.  For example, Jenkins triggers call back to `igor` to get information on the build that just completed; we'll now be asking igor for information on the completed build twice per pipeline.  Is that extra traffic/duplicated computation a concern?

Reply:
If it turns out we don't want all this extra computation on each trigger, one option would be to update `ManualEventHandler` to always fetch the pipeline being executed from `front50` before processing the pipeline.  This would solve the manual execution case (which I think is the main goal) without doing any re-computation.

Reply:
Ah thanks for pointing this out, I didn't realize `withMatchingTrigger` would make more external calls. 

Putting that logic back in `ManualEventHandler` could work (that's where I initially focused) but this seemed like a nice generic place to perform the refresh. Let me think some more about it and come back with something.

Reply:
I tried a more general rework but I find the artifacts stuff and igor lookups difficult to grok. I am temporarily admitting defeat and added the refresh on manual triggers only for now, in the interest of fixing the burning issue quickly. @ezimanyi does 9ca5c486b9729c57e5c9da10441efb065e5fd6db look alright to you?

Reply:
Yes, that looks great, thanks!